multicore_runtime/cloud_logging.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2015 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""Provide a handler to log to Cloud Logging in JSON."""
+
+import json
+import logging
+import math
+import os
+
+LOG_PATH_TEMPLATE = '/var/log/app_engine/app.{}.json'
```

Comment:
Suggestion: use `app.{pid}.json` instead of an unnamed field, and then `LOG_PATH_TEMPLATE.format(pid=os.getpid())`. That would make it a lot more clear what is getting substituted here. Also allows you to extend it to take more fields and have the format mix them however it wants to.